src/libpmemobj/ulog.c

```diff
@@ -400,6 +400,19 @@ ulog_entry_val_create(struct ulog *ulog, size_t offset, uint64_t *dest,
 	return e;
 }
 
+/*
+ * ulog_clobber_entry -- zeroes out a single log entry header
+ */
+void
+ulog_clobber_entry(const struct ulog_entry_base *e,
+	const struct pmem_ops *p_ops)
+{
+	VALGRIND_ADD_TO_TX(e, CACHELINE_SIZE);
+	pmemops_memset(p_ops, (char *)e, 0, CACHELINE_SIZE,
+		PMEMOBJ_F_MEM_NONTEMPORAL);
+	VALGRIND_REMOVE_FROM_TX(e, CACHELINE_SIZE);
```

Comment:
I think all those CACHELINE_SIZEs should be replaced by sizeof(struct ulog).

Reply:
In this case it should be `ulog_entry_size(e);` or `CACHELINE_ALIGN(sizeof(struct ulog_entry_buf))`. The former won't work because it requires `e` to point to a valid entry, and the latter will work but seems unnecessary - entries have always at least one cacheline.

Reply:
I think Piotrek is right, because we need just zero offset in the ulog entry base.